examples/clients/clients.bzl

```diff
@@ -0,0 +1,53 @@
+def swagger_codegen(name, spec, package_name, outputs, out_dir=".", **kwarg):
+  out_dir_spec = "$(@D)"
+  if out_dir != ".":
+    out_dir_spec += "/" + out_dir
+    for out in outputs:
+      if not out.startswith(out_dir + "/"):
+        fail("%s must reside in the out_dir %s" % (out, out_dir), "outputs")
+
+  native.genrule(
+      name = name,
+      srcs = [spec],
+      outs = outputs,
+      cmd = " ".join([
+          "$(location //examples/clients:swagger-codegen-cli)",
```

Comment:
I wrote something very similar to this for the project I'm working on. One thing I did that turned out to be extremely useful was to generate into a temp dir, then move the files one by one into out_dir_spec, then rm the files that are generated by swagger that we don't care about, then rmdir the temporary dir.

It seems like a lot of additional work, and it is, but it provides you with a compile time error if you missed one of the generated files in your `srcs` list. WDYT about doing that?